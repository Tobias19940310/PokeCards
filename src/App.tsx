import {useEffect, useState} from 'react';
import {Container} from "@material-ui/core";

import NavBar from "./components/NavBar";
import Content from "./components/Content";

import {baseAllPokemon} from "./data/baseItems";
import {baseSinglePokemon} from "./data/baseItems";
import {IAllPokemon, ISinglePokemon} from "./data/InterfacesPokemon";
import {getAllPokemon, getSinglePokemon} from "./api/pokeApi";


function App() {

  const [allPokemon, setAllPokemon] = useState<IAllPokemon>(baseAllPokemon);
  const [singlePokemon, setSinglePokemon] = useState<ISinglePokemon>(baseSinglePokemon);
  const perPageLimit :number = 60;

  useEffect(() => {
    retrieveAllPokemon(0);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  //PASS OFFSET (AT WHICH ID DOES THE PAGE START)
  const retrieveAllPokemon = (offset:number) :void => {
    const url :string = `https://pokeapi.co/api/v2/pokemon/?offset=${offset}&limit=${perPageLimit}`
    getAllPokemon(url)
    .then((response :IAllPokemon) => {
      setAllPokemon(response);
    })
  }

  //PASS URL OF SINGLE POKEMON
  const retrieveSinglePokemon = (url:string) :void => {
    getSinglePokemon(url)
    .then((response :ISinglePokemon) => {
      setSinglePokemon(response);
    })
  }

  return (
    <Container id="App" maxWidth={false} data-testid="App">
      <NavBar />
      <Content 
        allPokemon={allPokemon} retrieveAllPokemon={retrieveAllPokemon} 
        perPageLimit={perPageLimit} 
        singlePokemon={singlePokemon} retrieveSinglePokemon={retrieveSinglePokemon}
        />
    </Container>
  );
}

export default App;

