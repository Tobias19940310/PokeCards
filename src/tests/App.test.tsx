import ReactDOM from 'react-dom';
import App from '../App';
import renderer from 'react-test-renderer';
import TestRenderer  from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
});

it('default Render Snapshot', () => {
    const tree = renderer
        .create(<App />)
        .toJSON();
    expect(tree).toMatchSnapshot();
    });