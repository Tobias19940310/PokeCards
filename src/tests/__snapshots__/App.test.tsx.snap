// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default Render Snapshot 1`] = `
<div
  className="MuiContainer-root"
  data-testid="App"
  id="App"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appbar-1 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="MuiBox-root MuiBox-root-14 makeStyles-header-2"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5"
        >
          PokeCards
        </h5>
      </div>
      <div
        className="MuiAvatar-root MuiAvatar-square makeStyles-logo-3"
      >
        <img
          className="MuiAvatar-img"
          src="pokedex_sm.png"
        />
      </div>
    </div>
  </header>
  <div
    className="MuiGrid-root makeStyles-container-5 MuiGrid-container"
  >
    <div
      className="MuiGrid-root makeStyles-center-6 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-5 MuiGrid-grid-md-4"
    >
      <div
        className="MuiPaper-root MuiCard-root makeStyles-grow-9 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          className="MuiCardContent-root"
          data-testid="defaultCard"
          id="defaultCard"
        >
          <img
            alt="Pokedex"
            className="MuiCardMedia-root makeStyles-mediaPadding-12 MuiCardMedia-media MuiCardMedia-img"
            src="pokedex.png"
            title="Pokedex"
          />
          <hr
            className="MuiDivider-root makeStyles-divider-13"
          />
          <h5
            className="MuiTypography-root makeStyles-padding-10 MuiTypography-h5 MuiTypography-gutterBottom MuiTypography-alignCenter"
          >
            This is a PokeDex
          </h5>
          <p
            className="MuiTypography-root makeStyles-padding-10 MuiTypography-body2 MuiTypography-alignJustify"
          >
            Select a Pokemon to display every information you could ever need!
          </p>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-center-6 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-7"
    >
      <div
        ata-testid="loadingPokemon"
        className="MuiCircularProgress-root makeStyles-spinner-8 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        id="loadingPokemon"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
